### ISSUE TITLE
Installing using composer require drupal/message_fos doesn't work

### ISSUE THREAD

**Issue by NewZeal** [mccrodp/message_fos#4]
The README suggests doing a composer update to install the third party plugins.  I do not want to do that because of all the unintended updates that I will get in my repo and so I use:  

`composer require drupal/message_fos`

Which is the suggested method for installing a new module.  The install does not work because of 

 ```
Could not find package drupal/message_fos at any version for your minimum-s  
  tability (dev). Check the package spelling or your minimum-stability     
```
I succeeded with the install using:

`composer require friendsofsymfony/message`

The documentation for FOS advise against using in production for now.  I'm not sure I want to use this package until it is further developed and could explain why FOS didn't install using the standard method.

**Comment by mccrodp** [comment]
Sure, maybe you're right. It's been a long time since I had this idea. Maybe that repo isn't being used as expected. Worth seeing if there's an alternative or learning from it.

Either way, I imagine its idea of "Conversations" like a wrapper entity is more future-proof and akin to Fb Messenger than a threaded approach.

I remember having a chat with `heylookalive` about comstack and it's similarities re: D7. He was using a Conversation approach, so might be worth taking a look there if interested too: https://www.drupal.org/project/comstack / https://www.drupal.org/project/comstack_pm. It's based on Message Stack also.

**Comment by NewZeal** [comment]
Yes, it would be nice to have a well developed third party messaging system.  I see that FOS has a message entity.  We already have that with the message module.  The thing we are missing is the thread/conversation entity.

My initial spec for the thread / conversation is as follows:
- User A creates a message and sends to User B and thread 1 is created
- User A and User B send messages to each other for a while, and they remain using thread 1
- User B adds User C to the send field.  On message submit he is asked if he wants to start a new conversation.  
  = If he agrees, then thread 2 begins and the conversation continues.  User C can only see thread 2.  The other two also have access to thread 1
  = If he says No, then it stays with thread 1 and User C can see from the beginning
- When a new thread is started from a previous thread, the relation between the threads is kept as well as the change event so that it can be reported: "User C entered the conversation"
- Related threads can be displayed concatenated, so that User A and B can see the whole conversation

For the inbox we display a list of threads in chrono order similar to privatemsg module.  
- Related threads are shown somehow in the list
- Users can see an entire thread by clicking on a suitable link

For the sent we only display messages that have received no reply.

I think I might be able to do all the above with one new entity.

I'm very much open to feedback on those suggestions.

**Comment by mccrodp** [comment]
Sounds good, but I would strip that down initially more inline to what Facebook has as a foundation. If A and B have a Conversation and C is added, they are added to that conversation. I they want to start a new conversation they can with the 3 participants. That lessens the logic required and is an easier starting point, the rest can come later as optional plugin or whatnot and be configurable by site builder / developer.

The "Sent" idea is interesting, maybe just one view with a few filters. Maybe "Unanswered" makes more sense as a string then, could be further filtered by "Sent" and "Received", etc.

**Comment by NewZeal** [comment]
That is a good point, but when you say "I they want to start a new conversation they can with the 3 participants" then that implies we have some kind of functionality to start a new conversation, which is the same amount of functionality that I am suggesting.  The difference with my suggestion is that we trigger that button only when the composition of the conversation changes.

Yes, we could rename the tabs to 'conversations' and 'unanswered' or similar.

**Comment by mccrodp** [comment]
You're right, it's similar, but I don't think equivalent:

>User B adds User C to the send field. On message submit he is asked if he wants to start a new conversation.
= If he agrees, then thread 2 begins and the conversation continues. User C can only see thread 2. The other two also have access to thread 1
= If he says No, then it stays with thread 1 and User C can see from the beginning

I'm stating remove this logic and let the user make these actions themselves, i.e. - not prompts initially, that could come later as it's not that essential (unless you need it for your project 😄 ).

i.e. - 

* User B adds User C to conversation (same conversation)
* User B selects "New Message" or "New Conversation" from the UI

What I mean is, if you look at Fb I don't think it has the logic you mention, no prompts, you can add to an existing conversation, or create a new one.
